src/store.cc

```diff
@@ -2029,13 +2029,24 @@ StoreEntry::detachFromDisk()
 void
 StoreEntry::checkDisk() const
 {
-    const bool ok = (swap_dirn < 0) == (swap_filen < 0) &&
-                    (swap_dirn < 0) == (swap_status == SWAPOUT_NONE) &&
-                    (swap_dirn < 0 || swap_dirn < Config.cacheSwap.n_configured);
-
-    if (!ok) {
+    try {
+        Must((swap_dirn < 0) == (swap_filen < 0));
+        Must((swap_dirn < 0) || (swap_dirn < Config.cacheSwap.n_configured));
+
+        if (swap_dirn < 0) {
+            Must(swap_status == SWAPOUT_NONE);
+        } else if (swap_status == SWAPOUT_FAILED) {
```

Comment:
Please move this check inside the `else` part of the `swap_dirn` `if` statement. Here is a rough sketch:

```
if (swap_dirn < 0) {
    Must(swap_status == SWAPOUT_NONE);
} else {
    Must(swap_dirn < Config.cacheSwap.n_configured);
    if (swap_status == SWAPOUT_FAILED)
        Must(EBIT_TEST(flags, RELEASE_REQUEST));
    else
        Must(swap_status == SWAPOUT_WRITING || swap_status == SWAPOUT_DONE);
}
```